prefq/examples/imitation_preference_comparisons.py

```diff
@@ -167,16 +168,23 @@ def _get_feedback_from_server(self):
         
         def _wait_for_feedback_request(url):
             while True:
-                time.sleep(5)
-                response = requests.get(url, timeout=5)
-                if response.status_code == 200:
-                    feedback_data = response.json()
-                    if feedback_data == {}:
-                        print("Query Client: Waiting for feedback...")
-                        continue
-                    else: 
-                        print("Query Client: Feedback received")
-                        break
+                try:
+                    time.sleep(5)
+                    response = requests.get(url, timeout=5)
+                    if response.status_code == 200:
+                        feedback_data = response.json()
+                        if feedback_data == {}:
+                            print("Query Client: Waiting for feedback...")
+                            continue
+                        else: 
+                            print("Query Client: Feedback received")
+                            break
+                except requests.exceptions.Timeout:
+                    print("Query Client: Timeout error occurred while waiting for feedback.")
+                    raise requests.exceptions.Timeout
```

Comment:
Similarly here: If you give the exception a name `except ... as exc` you can later re-raise it with `raise exc`.